src/objects/screen.c

```diff
@@ -781,73 +806,153 @@ static int luaC_screen_set_position(lua_State *L)
+
+    struct wlr_output_mode *mode;
+    lua_newtable(L);
+    int i = 1;
+    wl_list_for_each(mode, &output->wlr_output->modes, link)
+    {
+        lua_createtable(L, 3, 0);
+        lua_pushnumber(L,mode->width);
+        lua_rawseti(L,-2,1);
+        lua_pushnumber(L,mode->height);
+        lua_rawseti(L,-2,2);
+        lua_pushnumber(L,mode->refresh);
+        lua_rawseti(L,-2,3);
+        lua_rawseti(L,-2,i++);
+    }
+    return 1;
+}
+
+/** Set the screen mode to a custom mode.
+ *
+ * @method set_custom_mode
+ * @tparam integer width
+ * @tparam integer height
+ * @tparam integer refresh
+ * @noreturn
+ */
+static int luaC_screen_set_custom_mode(lua_State *L)
+{
+    struct cwc_output *output = luaC_screen_checkudata(L, 1);
+
+    int32_t width             = luaL_checkint(L, 2);
+    int32_t height            = luaL_checkint(L, 3);
+    int32_t refresh           = lua_tonumber(L, 4);
+
+    if(refresh < 1000){
+        refresh *= 1000;
+    }
+
+    wlr_output_state_set_custom_mode(&output->pending, width,height,refresh);
+    transaction_schedule_output(output);
+    return 0;
+}
+/** Set the screen mode from ID.
+ *
+ * @method set_mode_from_id
+ * @tparam integer index
+ * @treturn boolean success If change was successful
+ */
+static int luaC_screen_set_mode_from_id(lua_State *L)
+{
+    struct cwc_output *output = luaC_screen_checkudata(L, 1);
+    int32_t index             = luaL_checkint(L, 2);
+    
+    struct wlr_output_mode *mode;
+    bool found = false;
+    int i = 1;
+    wl_list_for_each(mode, &output->wlr_output->modes, link)
+    {
+        if (i++ == index) {
+            found = true;
+            break;
+        }
+    }
+    if (!found){
+        lua_pushboolean(L,false);
+        return 1;
+    }
+
+    wlr_output_state_set_mode(&output->pending, mode);
+    transaction_schedule_output(output);
+    lua_pushboolean(L,true);
+    return 1;
+}
+/** Set the screen mode using a provided mode by the screen.
  *
  * @method set_mode
  * @tparam integer width
  * @tparam integer height
  * @tparam[opt=0] integer refresh Monitor refresh rate in hz
- * @noreturn
+ * @treturn boolean success If change was successful
  */
 static int luaC_screen_set_mode(lua_State *L)
 {
     struct cwc_output *output = luaC_screen_checkudata(L, 1);
     int32_t width             = luaL_checkint(L, 2);
     int32_t height            = luaL_checkint(L, 3);
     int32_t refresh           = lua_tonumber(L, 4);
+    
+    if(refresh < 1000) refresh *= 1000;
 
     bool found = false;
     struct wlr_output_mode *mode;
     wl_list_for_each(mode, &output->wlr_output->modes, link)
     {
         if (mode->width == width && mode->height == height) {
-            int diff = abs(refresh - mode->refresh / 1000);
-            if (!refresh) {
-                found = true;
-                break;
-            } else if (diff >= 0 && diff <= 2) {
+            int diff = abs(refresh - mode->refresh);
```

Comment:
I think it should be reverted because in case there is an output with more than 1000hz user can't set it, also I want it to exclusively using hz to make it more uniform.

Reply:
Also everything from wlroots returns refresh rate multiplied by 0.

Reply:
Infact, if i remove the /1000 from cwctl's output, I get a bunch of modes that use non-rounded refresh rates
<img width="277" height="533" alt="image" src="https://github.com/user-attachments/assets/a6436733-b1d2-4f47-ad85-4a7ff8f563d5" />

Reply:
yeah sorry I was overthinking it.